### PR TITLE
Xpetra: fixes for ETI-off build

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/xpetra @trilinos/muelu

## Motivation
Fixes some issues with ETI=off. See #5729.

### CODE REVIEW

File: packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_decl.hpp [hunk 1]
```diff
@@ -56,6 +56,7 @@
 #include "MueLu_StructuredAggregationFactory_fwd.hpp"
 #include "MueLu_Level_fwd.hpp"
 #include "MueLu_Exceptions.hpp"
+#include "MueLu_AggregationStructuredAlgorithm_fwd.hpp"
```
Comment (lucbv) [inline packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_decl.hpp:59]: Why is this needed here when it's not referenced anywhere in the file? I would have assumed that it would go in the `_def` file instead?

File: packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_def.hpp [hunk 2]
```diff
@@ -49,7 +49,6 @@
 #include <Xpetra_Map.hpp>
 #include <Xpetra_CrsGraph.hpp>
 
-#include "MueLu_AggregationStructuredAlgorithm.hpp"
```
Comment (lucbv) [inline packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_def.hpp:52]: I guess same question as above?

**Review by lucbv — DISMISSED** [review]
lgtm but I have a couple questions...

File: packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_decl.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/muelu/src/Graph/StructuredAggregation/MueLu_StructuredAggregationFactory_decl.hpp:59]: I believe this is needed to make short names work.

File: packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp [hunk 3]
```diff
@@ -2955,10 +2956,11 @@ namespace MueLu {
     RCP<const Thyra::LinearOpBase<Scalar> > thyraA = Xpetra::ThyraUtils<Scalar,LocalOrdinal,GlobalOrdinal,Node>::toThyra(Teuchos::rcp_dynamic_cast<Xpetra::CrsMatrixWrap<Scalar,LocalOrdinal,GlobalOrdinal,Node>>(A)->getCrsMatrix());
 
     Stratimikos::DefaultLinearSolverBuilder linearSolverBuilder;
-    Stratimikos::enableMueLu<LocalOrdinal,GlobalOrdinal,Node>(linearSolverBuilder);
+    typedef Thyra::PreconditionerFactoryBase<Scalar>                                     Base;
+    typedef Thyra::MueLuPreconditionerFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node> ImplMueLu;
+    linearSolverBuilder.setPreconditioningStrategyFactory(Teuchos::abstractFactoryStd<Base, ImplMueLu>(), "MueLu");
```
Comment (jhux2) [inline packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2961]: Was this to address warnings?  The only error I saw was in the structured aggregation, but the browser window was a bit hard to navigate.

Comment (cgcgcg) [reply packages/muelu/adapters/xpetra/MueLu_RefMaxwell_def.hpp:2961]: I couldn't get it to work correctly with ETI=off, it bugged out saying that `enableMueLu` can't be found in Stratimikos. I finally gave up and copied the 3 lines of code over.

**Review by jhux2 — APPROVED** [review]
lgtm
